Add note on api compatibility (#862)
cert-manager/website: content/en/docs/contributing/crds.md

## Versions

cert-manager currently has a single `v1` API version.

> The API types are defined in [`//pkg/apis/certmanager`](https://github.com/jetstack/cert-manager/tree/master/pkg/apis/certmanager). ACME related resources are in [`//pkg/apis/acme`](https://github.com/jetstack/cert-manager/tree/master/pkg/apis/certmanager).

Reviewer: nit (non-blocking): Could we also change the URL to the new cert-manager repo here while we're at it? Not important though, we'll need to do that site-wide at some point.

> Code comments on API type fields are being converted into documentation on our website and text of `kubectl explain`.

Reviewer 2: fwiw, `are being` is generally awkward, it's passive. What's doing the conversion?

It might be as simple as:

```suggestion
Code comments on API type fields are converted into documentation on our website and text included in `kubectl explain`.
```

Reviewer: I like the proposed suggestion here, it reads better IMO.

## Kubebuilder

While cert-manager doesn't fully use Kubebuilder for everything CRDs can make use of special Kubebuilder flags such as [validation flags](https://book.kubebuilder.io/reference/markers/crd-validation.html). We recommend reading the Kubebuilder book to learn more about them!

## Making changes to API

> Please see our [API compatibility promise](../../installation/api_compatibility/) to understand what changes to API are acceptable- the gist of it is that new fields can be added, but not removed. This also means that when a field is added to a version of the API, it can no longer be removed and its name cannot be changed- so we try to be cautious when adding new fields. Same applies to [constants and enumerated types](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#enumerated-or-constant-values).

Reviewer: I think this is fine as-is, but I don't don't mind the suggestion; I'm neutral on this suggestion.
